### PR TITLE
Bump Iceberg from 1.10.1 to 1.11.0

### PR DESCRIPTION
**Pull request by nastra**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (Iceberg)

## Description

This bump Iceberg from 1.10.1 to 1.11.0 and also updates custom classes to their respective version from upstream

## How was this patch tested?
existing tests

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: project/ShadedIcebergBuild.scala [hunk 1]
```diff
@@ -71,7 +71,6 @@ object ShadedIcebergBuild {
    * Here's an overview:
    *  PartitionSpec: sets checkConflicts to false to honor field ID assigned by Delta
    *  HiveCatalog, HiveTableOperations: allow metadataUpdates to overwrite schema and partition spec
-   *  RESTFileScanTaskParser: fixes NoSuchElementException on empty delete-file-references arrays
```
Comment (ChengJiX) [inline project/ShadedIcebergBuild.scala:74]: Is this no longer needed? Does Iceberg fix it in 1.11.0?

Comment (nastra) [reply project/ShadedIcebergBuild.scala:74]: yes we fixed that upstream

File: iceberg/src/test/scala/org/apache/spark/sql/delta/serverSidePlanning/SparkToIcebergExpressionConverterSuite.scala [hunk 2]
```diff
@@ -563,7 +563,7 @@ class SparkToIcebergExpressionConverterSuite extends AnyFunSuite {
 
   test("type conversions (Date/Timestamp) and boundary values") {
     val testDate = java.sql.Date.valueOf("2024-01-01")
-    val expectedDateDays = (testDate.getTime / (1000L * 60 * 60 * 24)).toInt
+    val expectedDateDays = testDate.toLocalDate.toEpochDay.toInt
```
Comment (nastra) [inline iceberg/src/test/scala/org/apache/spark/sql/delta/serverSidePlanning/SparkToIcebergExpressionConverterSuite.scala:566]: this fails otherwise when running locally in a timezone different from UTC

File: project/plugins.sbt [hunk 3]
```diff
@@ -24,7 +24,7 @@ addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
 
 addSbtPlugin("com.github.sbt" % "sbt-unidoc" % "0.5.0")
 
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "2.1.0")
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "2.3.1")
```
Comment (nastra) [inline project/plugins.sbt:27]: this is needed because Iceberg bundles records and sbt-assembly 2.1.0 can't handle those

File: iceberg/src/test/java/shadedForDelta/org/apache/iceberg/rest/IcebergRESTCatalogAdapterWithPlanSupport.java [hunk 4]
```diff
@@ -307,25 +305,6 @@ private TableIdentifier extractTableIdentifier(String path) {
     return TableIdentifier.of(namespace, tableName);
   }
 
-  /**
-   * Extract min-rows-requested from JSON string using Jackson.
-   * Iceberg 1.11 added this field, but we're on 1.10.0, so we parse it from JSON.
-   */
-  private Long extractMinRowsRequested(String jsonBody) {
```
Comment (nastra) [inline iceberg/src/test/java/shadedForDelta/org/apache/iceberg/rest/IcebergRESTCatalogAdapterWithPlanSupport.java:314]: no longer needed, because 1.11.0 has a method for retrieving that

**Review by tdas — APPROVED** [review]
build changes LGTM. Please rely on others approvals before merging this Pr.

**Review by harperjiang — APPROVED** [review]

**Review by ChengJiX — APPROVED** [review]
LGTM

**Review by murali-db — APPROVED** [review]
